python_modules/libraries/dagster-duckdb-pyspark/dagster_duckdb_pyspark/duckdb_pyspark_type_handler.py

```diff
@@ -211,3 +213,12 @@ def type_handlers() -> Sequence[DbTypeHandler]:
     @staticmethod
     def default_load_type() -> Optional[Type]:
         return pyspark.sql.DataFrame
+
+
+# This is temporarily needed until pyspark supports converting timestamps properly to pandas 2.x.
+# See: https://stackoverflow.com/questions/76072664/convert-pyspark-dataframe-to-pandas-dataframe-fails-on-timestamp-column
+def _pyspark_df_to_pandas_df(df: pyspark.sql.DataFrame) -> pd.DataFrame:
```

Comment:
@sryza Pinging you for confirmation that this is OK and non-destructive, since I don't know pyspark at all.

Basically the call `df.toPandas()` fails with pandas 2 if the spark df contains a timestamp column, because of either a bug or just failure of pyspark to fully support pandas 2-- this function is a workaround that changes any pyspark df timestamp columns before the call to `df.toPandas` so that they are properly autoconverted.

Reply:
duckdb also has support for direct ingestion from polars and pyarrow now (they didn't when this lib was written) so we could convert to a different table type if that makes more sense https://duckdb.org/docs/api/python/overview#dataframes

Reply:
I DMed sandy about this and he didn't know the answer. Then I investigated a bit more and I'm pretty sure this is not a clean conversion, it is ending up in duckdb as a string column instead of a timestamp.

> duckdb also has support for direct ingestion from polars and pyarrow now

Do you know if we can easily convert to one of these intermediate formats from pyspark (equivalent of `.toPandas()`)?

Reply:
@jamiedemaria thanks for your help, I got it to work by converting to arrow instead of pandas